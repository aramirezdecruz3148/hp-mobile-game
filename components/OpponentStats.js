import React from "react";
import { View, Text, Image } from "react-native";
import PropTypes from "prop-types";

export default function OpponentStats({ name, image, health }) {
  return (
    <View>
      <Image style={{width: 50, height: 50}} source={{ uri: image }} />
      <Text>{name}</Text>
      <Text>Health: {health}</Text>
    </View>
  );
}

OpponentStats.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  health: PropTypes.number.isRequired
};