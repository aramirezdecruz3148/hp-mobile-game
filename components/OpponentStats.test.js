import React from 'react';
import { shallow } from 'enzyme';
import OpponentStats from './OpponentStats';

describe('OpponentStats component', () => {
  it('renders OpponentStats', () => {
    const wrapper = shallow(<OpponentStats 
      name="Draco Malfoy" 
      image="https://vignette.wikia.nocookie.net/harrypotter/images/7/7e/Draco_Malfoy_TDH.png/revision/latest/scale-to-width-down/350?cb=20180116013508" 
      health={10}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});