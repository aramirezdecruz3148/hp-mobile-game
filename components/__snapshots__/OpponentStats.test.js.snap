// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpponentStats component renders OpponentStats 1`] = `
<Fragment>
  <View>
    <Image
      source={
        Object {
          "uri": "https://vignette.wikia.nocookie.net/harrypotter/images/7/7e/Draco_Malfoy_TDH.png/revision/latest/scale-to-width-down/350?cb=20180116013508",
        }
      }
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 100,
          "borderWidth": 1,
          "height": 80,
          "width": 80,
        }
      }
    />
  </View>
  <View>
    <Text>
      Draco Malfoy
      's Health: 
      10
    </Text>
  </View>
</Fragment>
`;
